lms/views/api/canvas/sync.py

```diff
@@ -68,8 +79,16 @@ def _get_canvas_groups(self):
                 self.group_set(),
             )
 
-        # If not grading return all the groups in the course so the teacher can toggle between them.
-        return self._canvas_api.group_category_groups(self.group_set())
+        try:
+            # If not grading return all the groups in the course so the teacher can toggle between them.
+            groups = self._canvas_api.group_category_groups(group_set_id)
+        except CanvasAPIError as canvas_api_error:
+            raise CanvasGroupSetNotFound(group_set=group_set_id) from canvas_api_error
```

Comment:
Do we actually know that the group set doesn't exist in the course, when `CanvasAPIError` is raised here? I think there might be several reasons why a `CanvasAPIError` might have been raised. We might want to inspect the `canvas_api_error` object